### ISSUE TITLE
Defining dimension reduction Element types

### ISSUE THREAD

**Issue by philippjfr** [ioam/holoviews#1173]
Currently the behavior when reducing the dimensionality of an Element e.g. by sampling or aggregating is a bit ill-defined. Certain Element types like the Raster types will drop down to a Curve type when you ``sample`` or ``reduce`` them, while on other types like HeatMap or Scatter3D it will simply throw an Exception. Since I'm working on implementing the Raster interface right now we should figure out how to handle this consistently. My suggestion would be that Elements define lower-dimensional views, here's what that might look like:

```python
Raster._views = {0: hv.BoxWhisker, 1: hv.Curve}
HeatMap._views = {0: hv.BoxWhisker, 1: hv.Scatter}
```

If no view type is defined it will simply assume the current type supports the lower-dimensionality.

**Comment by jbednar** [comment]
Having such reduction functionality is important. I'm confused about how the proposed syntax would work, though. Seems like the type depends on the precise operation, doesn't it? 0d sampling just gives a scalar, while 0d aggregation gives a BoxWhisker?

**Comment by philippjfr** [comment]
Yes, I was confused there, reduction operations will obviously produce scalars in the 0D case, so ignore that, it's mostly the 2D->1D cases that need to be handled.

**Comment by philippjfr** [comment]
Unfortunately it gets more complicated. When you index a single value dimension on an ``RGB``, ``HSV`` Element it currently drops down to an ``Image``. This would also need to be captured.

**Comment by philippjfr** [comment]
And another case, when indexing a single coordinate on RGB/HSV types you're left with multiple value dimensions, not just a single scalar.

**Comment by jlstevens** [comment]
> And another case, when indexing a single coordinate on RGB/HSV types you're left with multiple value dimensions, not just a single scalar.

How did we handle it before? One thing you can imagine is returning a numeric tuple - this could be a generic way of returning multiple vdims that have been reduced to scalars. I'm not entirely sure if this is a good idea though...

**Comment by philippjfr** [comment]
We used to return ItemTables here but at some point consistently switched to regular Tables in cases like this.
